v4/newrelic/segments.go

```diff
@@ -214,18 +216,58 @@ func (s *DatastoreSegment) End() {
 	if s.StartTime.isEnded() {
 		return
 	}
+
+	s.addAttributes(s.StartTime.Span.SetAttribute)
 	s.StartTime.Span.SetName(s.name())
 	s.StartTime.end()
 }
 
+func (s *DatastoreSegment) addAttributes(set func(string, interface{})) {
```

Comment:
Passing a `set` func here as an arg made testing easier. It also made the code easier to read.

Reply:
I like this, but I think the name could change.  I see "set" and my first thought was of the data collection type of set, rather than a setter.  Would `setter` be an acceptable name for this parameter to you? 

Then again, I can also accept that I might be abnormal that I thought of a data structure rather than a setter, so if you think that name is better I'm okay with it.

Reply:
Works for me!